runtime/src/accounts_db.rs

```diff
@@ -3396,30 +3408,81 @@ impl AccountsDb {
             .report("external_purge_slots_stats", Some(1000));
     }
 
-    // TODO: This is currently:
-    // 1. Unsafe with scan because it can remove a slot in the middle
-    // of a scan.
-    // 2. Doesn't handle cache flushes that happen during the slot deletion (see comment below).
-    pub fn remove_unrooted_slot(&self, remove_slot: Slot) {
-        if self.accounts_index.is_root(remove_slot) {
-            panic!("Trying to remove accounts for rooted slot {}", remove_slot);
-        }
-
-        // TODO: Handle if the slot was flushed to storage while we were removing the cached
-        // slot above, i.e. it's possible the storage contains partial version of the current
-        // slot. One way to handle this is to augment slots to contain a "version", That way,
-        // 1) We clean older versions via the natural clean() pipeline
-        // without having to call this function out of band.
-        // 2) This deletion doesn't have to block on scan
-        // Reads will then always read the latest version of a slot. Scans will also know
-        // which version their parents because banks will also be augmented with this version,
-        // which handles cases where a deletion of one version happens in the middle of the scan.
+    // TODO: This is currently unsafe with scan because it can remove a slot in the middle
```

Comment:
What is your plan for this now? Does this unsafeness exist today and you're just calling it out here or does this PR add this unsafeness? Is the unsafeness worth whatever this is fixing? I'm missing some context.
I'm confused with 'currently'. Prior to your change, in master? Or, 'currently' because of your change?

Reply:
Yeah, this unsafeness does exist today, but luckily this function isn't called anywhere by the validator 😃 

Yeah it's definitely worth fixing, and is addressed in this next PR here: https://github.com/solana-labs/solana/pull/17471, a peek into the future 👀